common/logger/logger.ts

```diff
@@ -0,0 +1,78 @@
+import { isCommandArg } from '../util/basic';
+import { configure, addLayout, getLogger as getlog4jsLogger, Logger as Log4jsLogger } from 'log4js';
+
+addLayout('json', function () {
+    return function (logEvent) {
+        const message = logEvent.data.shift();
+        const data = logEvent.data.shift();
+        return JSON.stringify({ ...logEvent, message, data });
+    };
+});
+
+let appenders = ['out'];
+if (process.env.LOG_FORMAT === 'json') {
+    appenders = ['outJson'];
+}
+
+configure({
+    appenders: {
+        out: { type: 'stdout', layout: { type: 'coloured' } },
+        outJson: { type: 'stdout', layout: { type: 'json' } },
+    },
+    categories: {
+        default: {
+            appenders,
+            level: isCommandArg('--debug') ? 'debug' : 'info',
+        },
+    },
+});
+
+export type LogData = { [key: string]: any };
+
+export class Logger {
+    constructor(private logger: Log4jsLogger) {}
+
+    getLoggerImpl(): any {
+        return this.logger;
+    }
+
+    trace(message: string, args: LogData = {}): void {
+        this.logger.trace(message, args);
+    }
+
+    // TODO switch to string
+    debug(message: any, args: LogData = {}): void {
+        if (typeof message !== 'string') {
+            message = JSON.stringify(message);
+        }
+        this.logger.debug(message, args);
+    }
```

Comment:
There are a looot of debug statements that just provide data without any message.
This is a shortcut for now and we have to check every single one in the future.